atxsdata/data.go

```diff
@@ -194,7 +194,7 @@ func NotMalicious(data *ATX, _ lockGuard) bool {
 // IterateInEpoch calls `fn` for every ATX in epoch.
 // If filters are provided, only atxs that pass all filters are returned.
 // SAFETY: The returned pointer MUST NOT be modified.
-func (d *Data) IterateInEpoch(epoch types.EpochID, fn func(types.ATXID, *ATX), filters ...AtxFilter) {
+func (d *Data) IterateInEpoch(epoch types.EpochID, fn func(*types.ATXID, *ATX), filters ...AtxFilter) {
```

Comment:
Are you certain that in cases like this it actually improves the performance? Passing by pointer could cause more GC pressure as opposed to passing by value, in which case the atxid would leave on the stack.

Reply:
Passing a value to a method doesn't necessarily mean that it won't escape to the heap. We'd have to read the escape analysis for the whole codebase to figure that out. If you pass a value it might end up on the heap anyway depending on what you do with it. So my consideration is ideally: "end up with one allocation of value" vs. "end up with possible multiple allocations".